lib/qx/tool/compiler/app/loader-server.tmpl.js

```diff
@@ -155,15 +155,9 @@
       },
 
       signalStartup : function() {
-        qx.Bootstrap.executePendingDefers();
-        qx.$$loader.delayDefer = false;
         qx.$$loader.scriptLoaded = true;
-        if (window.qx && qx.event && qx.event.handler && qx.event.handler.Application) {
-          qx.event.handler.Application.onScriptLoaded();
-          qx.$$loader.applicationHandlerReady = true;
-        } else {
-          qx.$$loader.applicationHandlerReady = false;
-        }
+        qx.core.BaseInit.ready();
+        qx.$$loader.applicationHandlerReady = true;
```

Comment:
`qx.event.handler.Application.onScriptLoaded();` is valid even for server apps because server apps derive from `qx.application.Basic`; however I think I recall that there needs to be a so fix that `qx.application.IApplication` has an `@require(qx.event.handler.Application)` that the first path of this "if" statement is activated properly.

If there isn't an application, then I can see that `qx.core.BaseInit.ready()` should be called, but that would be the `else` statement, not replacing the application startup

Reply:
Even if you patch qx.application.IApplication with @require(qx.event.handler.Application)  constructor qx.event.handler.Application is not called so that  $$instance is undefined. So onScriptLoaded did not work.

I just copied from the loader of the current framework

Reply:
That are nearly the same results i debugged yesterday. qx.core.BaseInit.ready is just called as result of an windows Event. So in server enviroment it must be called direct.